### ISSUE TITLE
connection crash while running explain (verbose, format json) select query

### ISSUE THREAD

**Issue by chenbaggio** [greenplum-db/gpdb#9359]
### Greenplum version or build
v6.0 stable
### OS version and uname -a
centos 
Linux computing-8c091891-master-6aadb510 3.10.0-957.21.3.el7.x86_64 #1 SMP Tue Jun 18 16:35:19 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

### Expected behavior
show query plan in json format  
### Actual behavior
crash with the stack:

```
#0  0x00007f225a07d49b in raise () from /lib64/libpthread.so.0
#1  0x0000000000b05e56 in StandardHandlerForSigillSigsegvSigbus_OnMainThread (processName=<optimized out>, postgres_signal_arg=11) at elog.c:5586
#2  <signal handler called>
#3  show_dispatch_info (slice=0x2ee97f0, es=es@entry=0x7ffd64153830, plan=0x2b91c58, plan=0x2b91c58) at explain.c:971
#4  0x00000000006b63b8 in ExplainNode (planstate=0x2af5b58, ancestors=ancestors@entry=0x2b320d0, relationship=relationship@entry=0xd37a0d "Inner", plan_name=plan_name@entry=0x0, 
    es=es@entry=0x7ffd64153830) at explain.c:1624
#5  0x00000000006b4c0d in ExplainNode (planstate=<optimized out>, ancestors=ancestors@entry=0x2b320d0, relationship=relationship@entry=0xd37a00 "Outer", plan_name=plan_name@entry=0x0, 
    es=es@entry=0x7ffd64153830) at explain.c:2291
#6  0x00000000006b64fe in ExplainNode (planstate=<optimized out>, ancestors=0x2b320d0, ancestors@entry=0x0, relationship=relationship@entry=0x0, plan_name=plan_name@entry=0x0, es=es@entry=0x7ffd64153830)
    at explain.c:2278
#7  0x00000000006b74f5 in ExplainPrintPlan (es=es@entry=0x7ffd64153830, queryDesc=queryDesc@entry=0x2ee8758) at explain.c:775
#8  0x00007f224b8e092b in acquire_query_plan (hdss_ri=0x7ffd64153890, es=0x7ffd64153830, queryDesc=0x2ee8758) at hashdata_stat_statements.c:830
#9  hdss_ExecutorEnd (queryDesc=0x2ee8758) at hashdata_stat_statements.c:656
#10 0x00000000006cd725 in PortalCleanup (portal=0x1f58648) at portalcmds.c:325
#11 0x0000000000b3a95a in PortalDrop (portal=0x1f58648, isTopCommit=<optimized out>) at portalmem.c:535
#12 0x00000000009a0324 in exec_simple_query (
    query_string=query_string@entry=0x1ce1358 "SELECT '' AS six, f1 AS \"Uncorrelated Field\" FROM SUBSELECT_TBL\n  WHERE f1 IN (SELECT f2 FROM SUBSELECT_TBL WHERE\n    f2 IN (SELECT f1 FROM SUBSELECT_TBL));", serializedLogicPartCtx=serializedLogicPartCtx@entry=0x0, serializedLogicPartCtxLen=serializedLogicPartCtxLen@entry=0) at postgres.c:1838
#13 0x00000000009a3033 in PostgresMain (argc=<optimized out>, argv=argv@entry=0x1c170a0, dbname=0x1c16f50 "postgres", username=<optimized out>) at postgres.c:5268
#14 0x0000000000544952 in BackendRun (port=0x1bf1920) at postmaster.c:4930
#15 BackendStartup (port=0x1bf1920) at postmaster.c:4587
#16 ServerLoop () at postmaster.c:2020
#17 0x00000000008806c5 in PostmasterMain (argc=argc@entry=6, argv=argv@entry=0x1be2bf0) at postmaster.c:1560
#18 0x0000000000546f62 in main (argc=6, argv=0x1be2bf0) at main.c:252
```
### Step to reproduce the behavior

1. run all the following SQLs

```
--
-- SUBSELECT
--

SELECT 1 AS one WHERE 1 IN (SELECT 1);

SELECT 1 AS zero WHERE 1 NOT IN (SELECT 1);

SELECT 1 AS zero WHERE 1 IN (SELECT 2);

-- Check grammar's handling of extra parens in assorted contexts

SELECT * FROM (SELECT 1 AS x) ss;
SELECT * FROM ((SELECT 1 AS x)) ss;

(SELECT 2) UNION SELECT 2;
((SELECT 2)) UNION SELECT 2;

SELECT ((SELECT 2) UNION SELECT 2);
SELECT (((SELECT 2)) UNION SELECT 2);

SELECT (SELECT ARRAY[1,2,3])[1];
SELECT ((SELECT ARRAY[1,2,3]))[2];
SELECT (((SELECT ARRAY[1,2,3])))[3];

-- Set up some simple test tables

CREATE TABLE SUBSELECT_TBL (
  f1 integer,
  f2 integer,
  f3 float
);

INSERT INTO SUBSELECT_TBL VALUES (1, 2, 3);
INSERT INTO SUBSELECT_TBL VALUES (2, 3, 4);
INSERT INTO SUBSELECT_TBL VALUES (3, 4, 5);
INSERT INTO SUBSELECT_TBL VALUES (1, 1, 1);
INSERT INTO SUBSELECT_TBL VALUES (2, 2, 2);
INSERT INTO SUBSELECT_TBL VALUES (3, 3, 3);
INSERT INTO SUBSELECT_TBL VALUES (6, 7, 8);
INSERT INTO SUBSELECT_TBL VALUES (8, 9, NULL);

SELECT '' AS eight, * FROM SUBSELECT_TBL;

-- Uncorrelated subselects

SELECT '' AS two, f1 AS "Constant Select" FROM SUBSELECT_TBL
  WHERE f1 IN (SELECT 1);

SELECT '' AS six, f1 AS "Uncorrelated Field" FROM SUBSELECT_TBL
  WHERE f1 IN (SELECT f2 FROM SUBSELECT_TBL);
```
2. run the following sql
```
set optimizer=off;
```
3.  run the following sql

```
explain (verbose, format json) SELECT '' AS six, f1 AS "Uncorrelated Field" FROM SUBSELECT_TBL
  WHERE f1 IN (SELECT f2 FROM SUBSELECT_TBL WHERE
    f2 IN (SELECT f1 FROM SUBSELECT_TBL));

```

after the three step, the issue is reproduced

**Comment by hlinnaka** [comment]
Thanks for the report! Opened https://github.com/greenplum-db/gpdb/pull/9369 to fix this.

**Comment by hlinnaka** [comment]
Fixed in commit 512578487b1df105f790dc554082eae4b543126a.
